once_test.py

```diff
@@ -86,9 +85,7 @@ def run():
         event.set()
         return func(*args, **kwargs)
 
-    future = executor.submit(run)
-    event.wait()
-    return future
+    return executor.submit(run)
```

Comment:
Regression here, just need to merge in the submitted PR I think

Reply:
done!